set-up-location-driven-geotriggers/src/main/java/com/esri/arcgismaps/sample/setuplocationdrivengeotriggers/MainActivity.kt

```diff
@@ -0,0 +1,374 @@
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.arcgismaps.ApiKey
+import com.arcgismaps.ArcGISEnvironment
+import com.arcgismaps.arcade.ArcadeExpression
+import com.arcgismaps.data.ArcGISFeature
+import com.arcgismaps.data.ServiceFeatureTable
+import com.arcgismaps.geometry.Geometry
+import com.arcgismaps.geometry.Polyline
+import com.arcgismaps.geotriggers.*
+import com.arcgismaps.location.LocationDataSourceStatus
+import com.arcgismaps.location.LocationDisplayAutoPanMode
+import com.arcgismaps.location.SimulatedLocationDataSource
+import com.arcgismaps.location.SimulationParameters
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.portal.Portal
+import com.arcgismaps.portal.PortalItem
+import com.esri.arcgismaps.sample.setuplocationdrivengeotriggers.databinding.ActivityMainBinding
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val playPauseFAB: FloatingActionButton by lazy {
+        activityMainBinding.playPauseFAB
+    }
+
+    private val sectionButton: Button by lazy {
+        activityMainBinding.sectionButton
+    }
+
+    // recycler list view to show the the points of interest
+    private val poiListView by lazy {
+        activityMainBinding.poiListView
+    }
+
+    // custom list adapter for the points of interest
+    private val poiListAdapter by lazy {
+        // create a new feature list adapter from the poiList
+        FeatureListAdapter(poiList) { feature ->
+            // set the item callback to show the feature view fragment
+            showFeatureViewFragment(feature)
+        }
+    }
+
+    private val simulatedLocationDataSource: SimulatedLocationDataSource by lazy {
+        // create SimulationParameters starting at the current time,
+        // a velocity of 10 m/s, and a horizontal and vertical accuracy of 0.0
+        val simulationParameters = SimulationParameters(
+            Clock.System.now(),
+            3.0,
+            0.0,
+            0.0
+        )
+        SimulatedLocationDataSource().apply {
+            // use the polyline from this ArcGIS Online GeoJSON to define the path. retrieved
+            // from https://https://arcgisruntime.maps.arcgis.com/home/item.html?id=2a346cf1668d4564b8413382ae98a956
+            setLocationsWithPolyline(
+                Geometry.fromJson(getString(R.string.polyline_json)) as Polyline,
+                simulationParameters
+            )
+        }
```

Comment:
There's a constructor for this so no need to use apply
```suggestion
        // use the polyline from this ArcGIS Online GeoJSON to define the path. retrieved
        // from https://https://arcgisruntime.maps.arcgis.com/home/item.html?id=2a346cf1668d4564b8413382ae98a956
        SimulatedLocationDataSource(
            Geometry.fromJson(getString(R.string.polyline_json)) as Polyline,
            simulationParameters
        )
```